src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/GaussianMixtureModel.java

```diff
@@ -0,0 +1,635 @@
+     */
+    public static GaussianMixtureModel makeEmptyModel(final int numGaussians, final int numAnnotations,
+                                                      final double prior_alpha, final double prior_beta, final double prior_nu) {
+        return new GaussianMixtureModel(numGaussians, numAnnotations, prior_alpha, prior_beta, prior_nu);
+    }
+
+    /**
+     * Fits a model to the provided data and returns a fitted model.
+     */
+    public static GaussianMixtureModel makeFittedModel( final List<VariantDatum> data, final int maxGaussians, final VariantRecalibratorArgumentCollection VRAC) {
+        if( data == null || data.isEmpty() ) { throw new IllegalArgumentException("No data found."); }
+        if( maxGaussians <= 0 ) { throw new IllegalArgumentException("maxGaussians must be a positive integer but found: " + maxGaussians); }
+
+        final GaussianMixtureModel model = GaussianMixtureModel.makeEmptyModel(maxGaussians, data.get(0).annotations.getDimension(), VRAC.DIRICHLET_PARAMETER, VRAC.SHRINKAGE, VRAC.PRIOR_COUNTS );
+        model.initializeRandomModel(data, VRAC.NUM_KMEANS_ITERATIONS, Utils.getApacheRandomGenerator());
+        model.variationalBayesExpectationMaximization(data, VRAC.MAX_ITERATIONS, MIN_PROB_CONVERGENCE);
+        return model;
+    }
+
+    /**
+     * Sets the gaussians to the arguments (removes the current one).
+     * Note that the actual data is not copied, just set.
+     * Strictly for testing only.
+     */
+    @VisibleForTesting
+    void setGaussians(Collection<MultivariateGaussian> mvgs) {
+        gaussians.clear();
+        gaussians.addAll(mvgs);
+    }
+
+    /**
+     * Returns the gaussians.
+     * Use with caution - the live objects are returned and must not be modified.
+     * Only for testing.
+     */
+    @VisibleForTesting
+    Collection<MultivariateGaussian> getGaussians() {
+        return gaussians;
+    }
+
+    /**
+     * Fits the Gaussian Mixture model to the data.
+     * The model is assumed to have been initialized (eg by k-means).
+     */
+    private void variationalBayesExpectationMaximization( final List<VariantDatum> data, int maxIterations, double minProbConvergence ) {
+        // The VBEM loop
+        normalizePMixtureLog10(data.size());
+        expectationStep(data);
+        double currentChangeInMixtureCoefficients;
+        int iteration = 0;
+        logger.info("Finished iteration " + iteration + ".");
+        while( iteration < maxIterations ) {
+            iteration++;
+            maximizationStep(data);
+            currentChangeInMixtureCoefficients = normalizePMixtureLog10(data.size());
+            expectationStep(data);
+            if( iteration % 5 == 0 ) { // cut down on the number of output lines so that users can read the warning messages
+                logger.info("Finished iteration " + iteration + ". \tCurrent change in mixture coefficients = " + String.format("%.5f", currentChangeInMixtureCoefficients));
```

Comment:
hmm. I guess the progresslogger is (currently) strictly for things that have a genomic position. not general iterations....perhaps this could be changed?